rcl_action/test/rcl_action/test_action_interaction.cpp

```diff
@@ -0,0 +1,906 @@
+
+  // Take result request with valid arguments
+  // rmw_request_id_t request_header;
+  ret = rcl_action_take_result_request(
+    &this->action_server, &request_header, &incoming_result_request);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Check that the result request was received correctly
+  EXPECT_TRUE(uuidcmp(
+      outgoing_result_request.uuid,
+      incoming_result_request.uuid));
+
+  // Initialize result response
+  ASSERT_TRUE(rosidl_generator_c__int32__Sequence__init(
+      &outgoing_result_response.sequence, 4));
+  outgoing_result_response.sequence.data[0] = 0;
+  outgoing_result_response.sequence.data[1] = 1;
+  outgoing_result_response.sequence.data[2] = 2;
+  outgoing_result_response.sequence.data[3] = 6;
+  outgoing_result_response.status =
+    action_msgs__msg__GoalStatus__STATUS_SUCCEEDED;
+
+  // Initialize cancel request
+  rmw_request_id_t cancel_request_header;
+  init_test_uuid0(outgoing_cancel_request.goal_info.goal_id.uuid);
+  outgoing_cancel_request.goal_info.stamp.sec = 321;
+  outgoing_cancel_request.goal_info.stamp.nanosec = 987654u;
+
+  // Send cancel request with valid arguments
+  ret = rcl_action_send_cancel_request(
+    &this->action_client, &outgoing_cancel_request, &sequence_number);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_wait_set_clear(&this->wait_set);
+
+  ret = rcl_action_wait_set_add_action_server(&this->wait_set, &this->action_server, NULL);
+  ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_wait(&this->wait_set, 1000000);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_action_server_wait_set_get_entities_ready(
+    &this->wait_set,
+    &this->action_server,
+    &this->is_goal_request_ready,
+    &this->is_cancel_request_ready,
+    &this->is_result_request_ready,
+    &this->is_goal_expired);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  EXPECT_TRUE(this->is_cancel_request_ready);
+  EXPECT_FALSE(this->is_goal_request_ready);
+  EXPECT_FALSE(this->is_result_request_ready);
+
+  // Take cancel request with valid arguments
+  ret = rcl_action_take_cancel_request(
+    &this->action_server, &cancel_request_header, &incoming_cancel_request);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Check that the cancel request was received correctly
+  EXPECT_TRUE(uuidcmp(
+      outgoing_cancel_request.goal_info.goal_id.uuid,
+      incoming_cancel_request.goal_info.goal_id.uuid));
+  EXPECT_EQ(
+    outgoing_cancel_request.goal_info.stamp.sec,
+    incoming_cancel_request.goal_info.stamp.sec);
+  EXPECT_EQ(
+    outgoing_cancel_request.goal_info.stamp.nanosec,
+    incoming_cancel_request.goal_info.stamp.nanosec);
+
+  rcl_action_cancel_request_t cancel_request = rcl_action_get_zero_initialized_cancel_request();
```

Comment:
`rcl_action_action_msgs__srv__CancelGoal_Request` is typedef'd to `rcl_action_cancel_request_t`, so I think this logic for doing the "conversion" is not necessary. I may be mistaken, but I think we can use `incoming_cancel_request` directly with `rcl_action_process_cancel_request`.  Note, the same is **not** true for `rcl_action_cancel_response_t`.

Reply:
You're right, it wasn't needed. Done!